### ISSUE TITLE
Unable to create customer through Admin on DemoSite

### ISSUE THREAD

**Issue by ckittrell** [BroadleafCommerce/BroadleafCommerce#1346]
Customer Care > Customer -----> Add

Result:
org.hibernate.id.IdentifierGenerationException: ids for this class must be manually assigned before calling save(): org.broadleafcommerce.profile.core.domain.CustomerImpl

**Comment by pbaggett** [comment]
@ckittrell I'm unable to reproduce this on latest DemoSite or DemoSitePrivate, on what version(s) is this occurring for you?

**Comment by ckittrell** [comment]
This was on DemoSite with `4.0.0-SNAPSHOT`

**Comment by pbaggett** [comment]
I'm building the latest DemoSite on the 4.0 branch, which uses blc version `4.0.0-SNAPSHOT`. My only modifications are to `maven.home` and `jrebel.path` in `build.properties`. I ran `mvn clean install`, then launched site and admin with `ant tomcat`.

Once running I logged into `http://localhost:8081/admin` with admin/admin. I clicked "Customer Care" -> "Customers", then "+ Add Customer", then "Save". The Customer was saved and I'm now at the edit Customer page for the Customer that was just created. I then repeated the process this time filling out "Email Address", "First Name", "Last Name", and selected "Registered: Yes", I end up at the Customer edit page for the Customer that was just created.

Is there anything that I'm missing in this process to trigger the error?
